modules/web/src/app/wizard/step/provider-settings/provider/extended/openstack/default/component.ts

```diff
@@ -275,6 +292,12 @@ export class OpenstackProviderExtendedCredentialsComponent
   ngOnDestroy(): void {
     this._unsubscribe.next();
     this._unsubscribe.complete();
+    this._networkChangeSubject.complete();
+  }
+
+  networkDisplayName(id: string): string {
+    const network: OpenstackNetwork = this.networks.find(network => network.id === id);
+    return network ? network.name || network.id : '';
```

Comment:
We should display the ID in `(ID: <network_id>)` format.

```suggestion
    return network ? network.name || `(ID: ${network.id})` : '';
```